examples/e12_custom_context.py

```diff
@@ -0,0 +1,69 @@
+from typing import TYPE_CHECKING, Any
```

Comment:
Can you move your example to just a `tests/utils/context/test_main.py` 2 tests? It is too complex to be an example